src/main/scala/strawman/collection/immutable/Set.scala

```diff
@@ -22,4 +23,15 @@ trait SetMonoTransforms[A, +Repr]
 
   def - (elem: A): Repr
 
+}
+
+trait SetPolyTransforms[A, +C[X] <: Set[X] with SetLike[X, C]]
+  extends collection.SetPolyTransforms[A, C] {
+
+  protected def coll: C[A]
+
+  def ++ (that: collection.Set[A]): C[A] =
+    if (that.isEmpty) coll
+    else that.foldLeft[C[A]](coll)(_ + _)
```

Comment:
Maybe the fold should be inlined into a bare iterator walk in `IterablePolyTransforms`.  Would that be faster?